### PR TITLE
www: Make Builder related URLs take buildername

### PR DESCRIPTION
**Pull request by tdesveaux**
This allow to share user-friendly URLs between users.

Note that using `buildername` in APIs can be problematic, see #7552.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `newsfragments` directory (and read the `README.txt` in that directory)
* [N/A] I have updated the appropriate documentation

### CODE REVIEW

**Comment by p12tic** [comment]
The problem with builder names in URLs is that builder names can change and then all URLs would break. I will think about how this could be solved. The PR itself looks good and would improve user experience.

**Comment by tdesveaux** [comment]
> The problem with builder names in URLs is that builder names can change and then all URLs would break. I will think about how this could be solved. The PR itself looks good and would improve user experience.

I agree that it can be an issue (in addition to #7552).
I looked into this as it was a feedback from users on things they missed since upgrading from 0.8.

But, maybe I missed something, but builderid seem to not be stable either in case of name change right?
I tried with the default sample from a `create-master`, ran a build then renamed `runtests` to `run-tests` and the BuilderId changed.
So wouldn't buildernames be as faulty as builderids in urls?

**Comment by p12tic** [comment]
>  So wouldn't buildernames be as faulty as builderids in urls?

You're completely right. Buildername has 1:1 relationship with builderid. Which means that there are no blockers for this PR.

**Comment by tdesveaux** [comment]
Still think merging this should be on hold until #7552 is addressed as this would add onto the inconsistency of which builder can be accessed by name.

**Comment by tdesveaux** [comment]
@p12tic since the issue with the data api and buildernames is fixed, there is no blocker for this AFAIK.

Let me know if you think of anything else.

File: www/base/src/views/BuilderView/BuilderView.tsx [hunk 1]
```diff
@@ -140,11 +141,11 @@ export const BuilderView = observer(() => {
   })));
 
   const builder = buildersQuery.getNthOrNull(0);
-  const builds = buildsQuery.getParentCollectionOrEmpty(builderid.toString());
-  const buildrequests = buildrequestsQuery.getParentCollectionOrEmpty(builderid.toString());
-  const forceschedulers = forceSchedulersQuery.getParentCollectionOrEmpty(builderid.toString());
+  const builds = buildsQuery.getParentCollectionOrEmpty(builder?.id ?? "");
```
Comment (p12tic) [inline www/base/src/views/BuilderView/BuilderView.tsx:144]: This introduces another roundtrip until contents can be displayed.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
Sorry for a long delay. I've been assuming that there is some risk. The risk didn't exist and I understood that only after taking a deep look into the PR for the n-th time.

**Pull request merged**